### PR TITLE
tikv: fix the transaction correctness of ResetToVersionWorker

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: ref #13276.

What's Changed:

```commit-message
This PR tries to fix the transaction correctness of `ResetToVersionWorker` without considering the case
that the TS to reset may be greater than the resolved TS and adds more test cases.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

File: src/server/reset_to_version.rs [hunk 1]
```diff
@@ -64,51 +100,57 @@ struct Batch {
...
         Ok(None)
     }
 
-    fn scan_next_batch(&mut self, batch_size: usize) -> Result<Batch> {
-        let mut writes = Vec::with_capacity(batch_size);
+    fn scan_next_batch(&mut self) -> Result<Batch> {
+        let mut writes = Vec::with_capacity(BATCH_SIZE);
         let mut has_more = true;
-        for _ in 0..batch_size {
+        for _ in 0..BATCH_SIZE {
```
Comment (Connor1996) [inline src/server/reset_to_version.rs:151]: while writes.len() < BATCH_SIZE

File: src/server/reset_to_version.rs [hunk 2]
```diff
@@ -120,56 +162,89 @@ impl ResetToVersionWorker {
...
+                            deleted: _
+                        }
+                    ));
+                    *state.scanned() += 1;
+                    self.total_scanned += 1;
+                }
+                let key = self.lock_iter.key();
+                self.write_batch.delete_cf(CF_LOCK, key)?;
+                deleted += 1;
+                let lock = box_try!(Lock::parse(self.lock_iter.value()));
```
Comment (Connor1996) [inline src/server/reset_to_version.rs:222]:
```suggestion
                let lock = Lock::parse(self.lock_iter.value())?;
```

**Comment by nolouch** [comment]
Does it fix the https://github.com/tikv/tikv/issues/13203?

**Comment by JmPotato** [comment]
> Does it fix the #13203?

I am not quite sure what bug causes it, but the current implementation of `ResetToVersionWorker` has several correctness problems. Let's fix the basic implementation first and re-test it fully again later.

File: src/server/reset_to_version.rs [hunk 3]
```diff
@@ -218,41 +294,60 @@ impl ResetToVersionManager {
...
+                                    break;
+                                }
+                            }
+                            Err(e) => {
+                                error!("Reset to version failed when resetting lock cf"; "error" => ?e);
+                                break;
+                            }
+                        }
+                    }
+                    info!("Reset to version done!"; "total_scanned" => worker.total_scanned);
```
Comment (Connor1996) [inline src/server/reset_to_version.rs:342]: print out the ts

Comment (Connor1996) [reply src/server/reset_to_version.rs:342]: print out how many keys are deleted for default/write/lock

**Review by Connor1996 — APPROVED** [review]

**Review by longfangsong — APPROVED** [review]

**Pull request closed without merging**